### PR TITLE
Support for Microsoft SQL Server [JDBC]

### PR DESCRIPTION
**Pull request by Sanne**
I think this is ready to be used.

There are some follow up things to discuss, but I don't think these should block merging the feature:
 - how to better document JDBC drivers; e.g. where do we want some examples and describe known limitations.
 - how to automatically enable the right character sets, as native-image only includes the essential ones.

### CODE REVIEW

File: extensions/jdbc/jdbc-mssql/jdbc-mssql-deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * @author Sanne Grinovero <sanne@hibernate.org>
+ */
+public final class MsSQLReflections {
+
+    @BuildStep
+    void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        //Not strictly necessary when using Agroal, as it also registers
+        //any JDBC driver being configured explicitly through its configuration.
+        //We register it for the sake of people not using Agroal.
+        final String driverName = "com.microsoft.sqlserver.jdbc.SQLServerDriver";
```
Comment (gsmet) [inline extensions/jdbc/jdbc-mssql/jdbc-mssql-deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java]: Can't you use the class itself to get the name? That would seem more future proof.

File: extensions/jdbc/jdbc-mssql/jdbc-mssql-runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        </dependency>
+        <!--
+            adal4j isn't used directly but you need to have it around for compilation to succeed.
+            Since that's the only reason, doesn't seem necessary to require the specific version in the shared bom.
+            -->
+        <dependency>
+            <groupId>com.microsoft.azure</groupId>
+            <artifactId>adal4j</artifactId>
+            <version>1.6.3</version>
+            <exclusions>
```
Comment (gsmet) [inline extensions/jdbc/jdbc-mssql/jdbc-mssql-runtime/pom.xml]: If you want to exclude all its dependencies, you can use:
```
<groupId>*</groupId>
<artifactId>*</artifactId>
```

**Review by gsmet — APPROVED** [review]
Nice work, glad to see this one tackled! I added 2 minor comments inline.

File: extensions/jdbc/jdbc-mssql/jdbc-mssql-deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java (hunk 1 above)
Comment (Sanne) [reply extensions/jdbc/jdbc-mssql/jdbc-mssql-deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java]: Tradeoff choices.

it's not like they can change the driver name without breaking a million apps either.. I prefer to not initialize the class.

File: extensions/jdbc/jdbc-mssql/jdbc-mssql-runtime/pom.xml (hunk 2 above)
Comment (Sanne) [reply extensions/jdbc/jdbc-mssql/jdbc-mssql-runtime/pom.xml]: unfortunately it wasn't possible to exclude them all :(

**Review by gsmet — APPROVED** [review]

**Pull request merged**